### PR TITLE
Upgrade to ESLint 9

### PR DESCRIPTION
**Pull request by jthemphill**
* Resolves https://github.com/tchajed/botc-tools/issues/13

Also fixes two lint warnings that I got after upgrading.

### CODE REVIEW

File: src/js/randomizer/tokens/token_canvas.ts [hunk 1]
```diff
@@ -19,7 +19,6 @@ function splitLinesCircle(
     const lineWidth = circleWidthAt(radius, radius - y) * widthFraction;
     // the current line is text.slice(0, i);
     let i = -1;
-    // eslint-disable-next-line no-constant-condition
```
Comment (jthemphill) [inline src/js/randomizer/tokens/token_canvas.ts:22]: > Unused eslint-disable directive (no problems were reported from 'no-constant-condition').

File: src/js/botc/roles.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ const RoleTypes = [
 ] as const;
 export type RoleType = (typeof RoleTypes)[number];
 
-const Editions = ["tb", "snv", "bmr", "other"] as const;
+export const Editions = ["tb", "snv", "bmr", "other"] as const;
```
Comment (jthemphill) [inline src/js/botc/roles.ts:39]: > `'Editions'` is assigned a value but only used as a type. Allowed unused vars must match `/^_/u`. [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)

File: package.json [hunk 3]
```diff
@@ -36,12 +37,11 @@
     "@types/react": "^18.2.7",
     "@types/react-dom": "^18.2.4",
     "@types/sharp": "^0.31.0",
-    "@typescript-eslint/eslint-plugin": "^7.1.0",
-    "@typescript-eslint/parser": "^7.1.0",
```
Comment (jthemphill) [inline package.json:40]: I had to import `typescript-eslint` to get at the recommended configs in the modern flat format, and these packages are part of `typescript-eslint`

File: eslint.config.mjs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import eslint from "@eslint/js";
+import tseslint from "typescript-eslint";
+import unusedImports from "eslint-plugin-unused-imports";
+import tsParser from "@typescript-eslint/parser";
+
+export default [
+  eslint.configs.recommended,
+  ...tseslint.configs.recommended,
```
Comment (jthemphill) [inline eslint.config.mjs:8]: Copied from https://typescript-eslint.io/getting-started/

File: src/js/randomizer/tokens/token_canvas.ts (hunk 1 above)
Comment (tchajed) [reply src/js/randomizer/tokens/token_canvas.ts:22]: That default is sensible and I don't think we need to make it explicit in the config

File: src/js/botc/roles.ts (hunk 2 above)
Comment (tchajed) [reply src/js/botc/roles.ts:39]: This seems fine; we could inline it but that seems uglier than having the definition available if needed, just like RoleTypes and RoleType.

**Comment by tchajed** [comment]
This looks great, thanks for doing that migration!

**Pull request merged**